### PR TITLE
Adding taxdump format and taxdump-to-tree converter

### PR DESCRIPTION
**Pull request by qiyunzhu**
Per discussion with @wasade , I created this format, parser and converter. It is useful for reading NCBI taxonomy database (`taxdump`) into data frames and convert them into a phylogenetic tree (`skbio.TreeNode`). It is potentially useful for level-flexible classification of microbiome data.

This functionality was inspired by a similar function implemented by me in Woltka: https://github.com/qiyunzhu/woltka. In developing the current PR, I re-wrote the code in a way that is coherent with scikit-bio standards.

The tentative usage is:

```python
import pandas as pd
from skbio import io, TreeNode
nodes = io.read('nodes.dmp', format='taxdump', scheme='nodes', into=pd.DataFrame)
names = io.read('names.dmp', format='taxdump', scheme='names', into=pd.DataFrame)
tree = TreeNode.from_taxdump(nodes, names)
```

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [ ] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: skbio/tree/_tree.py [hunk 1]
```diff
@@ -3269,3 +3274,116 @@ def unpack_by_func(self, func):
...
+            raise ValueError('There is no top-level node.')
+        elif n_top > 1:
+            raise ValueError('There are more than one top-level node.')
+
+        # get root taxid
+        root_id = tops.index[0]
+
+        # get parent-to-child(ren) map
+        to_children = {p: g.index.tolist() for p, g in nodes[
+            ~(nodes.index == root_id)].groupby('parent_tax_id')}
```
Comment (wasade) [inline skbio/tree/_tree.py:3352]:
```suggestion
            nodes.index != root_id].groupby('parent_tax_id')}
```

**Review by wasade — CHANGES_REQUESTED** [review]
This seems reasonable but would it be possible to not include the changes to the .c files?

**Comment by qiyunzhu** [comment]
@wasade I didn't touch the .c files. Perhaps the changes happen automatically when I ran unit tests? Do you know how to fix?

> This seems reasonable but would it be possible to not include the changes to the .c files?

**Comment by qiyunzhu** [comment]
> @wasade I didn't touch the .c files. Perhaps the changes happen automatically when I ran unit tests? Do you know how to fix?
> 
> > This seems reasonable but would it be possible to not include the changes to the .c files?

Manually fixed. No worry.

**Review by wasade — APPROVED** [review]

**Comment by wasade** [comment]
@qiyunzhu, I'm +1 on this. Could you add a mention in the changelog? We can merge after

**Comment by qiyunzhu** [comment]
@wasade Thanks! Changelog edited.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # scikit-bio changelog
 
+## Version 0.5.8
```
Comment (wasade) [inline CHANGELOG.md:3]:
```suggestion
## Version 0.5.8-dev
```

**Comment by wasade** [comment]
Thanks!!

**Pull request merged**